src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -345,16 +346,9 @@ public function doRequest(object $request): \Symfony\Component\BrowserKit\Respon
             $response = $app->handleRequest($yiiRequest);
             $app->trigger($app::EVENT_AFTER_REQUEST);
             $response->send();
-        } catch (\Exception $e) {
-            if ($e instanceof UserException) {
-                // Don't discard output and pass exception handling to Yii to be able
-                // to expect error response codes in tests.
-                $app->errorHandler->discardExistingOutput = false;
-                $app->errorHandler->handleException($e);
-            } elseif (!$e instanceof ExitException) {
```

Comment:
Was it necessary to remove this logic?

Reply:
No, see my comment in the issue itself. I am currently doubting whether the issue is valid at all.